### PR TITLE
Support s-shaped cross staff slurs

### LINKED ISSUE #1583: Support s-shaped slurs
When `slur/@curvedir="mixed"`, the rendering should be s-shaped like this:

grafik

```xml
<?xml version="1.0" encoding="UTF-16" ?>
<mei xml:id="m-1" meiversion="4.0.1" xmlns="http://www.music-encoding.org/ns/mei" xmlns:xlink="http://www.w3.org/1999/xlink">
<meiHead xml:id="m-2">
    <fileDesc xml:id="m-3">
        <titleStmt xml:id="m-4">
            <title xml:id="m-14"/>
        </titleStmt>
        <pubStmt xml:id="m-17">
            <availability xml:id="m-18">
                <useRestrict xml:id="m-19">Copyright © </useRestrict>
            </availability>
        </pubStmt>
    </fileDesc>
    <encodingDesc xml:id="m-5">
        <appInfo xml:id="m-6">
            <application xml:id="sibelius" isodate="2020-7-27T18:11:47Z" version="19900">
                <name xml:id="m-8" type="operating-system">Windows 8</name>
            </application>
            <application xml:id="sibmei" type="plugin" version="4.0.0">
                <name xml:id="m-10">Sibelius to MEI 4 Exporter (4.0.0)</name>
            </application>
        </appInfo>
    </encodingDesc>
    <workList xml:id="m-11">
        <work xml:id="m-12">
            <title xml:id="m-13"/>
        </work>
    </workList>
</meiHead>
<music xml:id="m-21">
    <body xml:id="m-22">
        <mdiv xml:id="m-23">
            <score xml:id="m-25">
                <scoreDef xml:id="m-26" lyric.name="Times New Roman" meter.count="4" meter.unit="4" music.name="Opus Std" page.botmar="12.7mm" page.height="297mm" page.leftmar="12.7mm" page.rightmar="12.7mm" page.topmar="12.7mm" page.width="210mm" ppq="256" spacing.staff="11" spacing.system="18.5" text.name="Times New Roman" vu.height="0.875mm">
                    <staffGrp xml:id="m-27">
                        <staffGrp xml:id="m-38" n="1" symbol="brace">
                            <staffDef xml:id="m-28" clef.line="2" clef.shape="G" key.mode="major" key.sig="0" lines="5" n="1">
                                <label xml:id="m-29">Piano</label>
                                <labelAbbr xml:id="m-30">Pno.</labelAbbr>
                                <!-- keyboard.piano.grand -->
                                <instrDef xml:id="m-32" midi.channel="1" midi.pan="53" midi.volume="100"/>
                            </staffDef>
                            <staffDef xml:id="m-33" clef.line="4" clef.shape="F" key.mode="major" key.sig="0" lines="5" n="2">
                                <label xml:id="m-34">Piano</label>
                                <labelAbbr xml:id="m-35">Pno.</labelAbbr>
                                <!-- keyboard.piano.grand -->
                                <instrDef xml:id="m-37" midi.channel="1" midi.pan="53" midi.volume="100"/>
                            </staffDef>
                        </staffGrp>
                    </staffGrp>
                </scoreDef>
                <section xml:id="m-39">
                    <pb xml:id="m-20"/>
                    <sb xml:id="m-41"/>
                    <measure xml:id="m-40" label="1" n="1">
                        <staff xml:id="m-42" n="1">
                            <layer xml:id="m-43" n="1">
                                <mRest xml:id="m-44" tstamp.real="00:00:00" visible="false"/>
                            </layer>
                        </staff>
                        <staff xml:id="m-45" n="2">
                            <layer xml:id="m-46" n="1">
                                <note xml:id="m-47" dur="4" oct="4" pname="c"/>
                                <note xml:id="m-48" dur="4" oct="3" pname="b"/>
                                <note xml:id="m-49" dur="4" oct="4" pname="c" staff="1"/>
                                <note xml:id="m-50" dur="4" oct="4" pname="d" staff="1"/>
                            </layer>
                        </staff>
                        <slur curvedir="mixed" xml:id="m-51" endid="#m-50" layer="1" staff="2" startid="#m-47"/>
                    </measure>
                    <measure xml:id="m-52" n="2">
                        <staff xml:id="m-53" n="1">
                            <layer xml:id="m-54" n="1">
                                <note xml:id="m-55" dur="4" oct="4" pname="c"/>
                                <note xml:id="m-56" dur="4" oct="4" pname="d"/>
                                <note xml:id="m-57" dur="4" oct="4" pname="c" staff="2"/>
                                <note xml:id="m-58" dur="4" oct="3" pname="b" staff="2"/>
                            </layer>
                        </staff>
                        <staff xml:id="m-60" n="2">
                            <layer xml:id="m-61" n="1">
                                <mRest xml:id="m-62" tstamp.real="00:00:02.4" visible="false"/>
                            </layer>
                        </staff>
                        <slur curvedir="mixed" xml:id="m-59" endid="#m-58" layer="1" staff="1" startid="#m-55"/>
                    </measure>
                    <measure xml:id="m-63" n="3" right="end">
                        <staff xml:id="m-64" n="1">
                            <layer xml:id="m-65" n="1">
                                <note xml:id="m-66" dur="1" oct="5" pname="c"/>
                            </layer>
                        </staff>
                        <staff xml:id="m-67" n="2">
                            <layer xml:id="m-68" n="1">
                                <beam xml:id="m-70">
                                    <note xml:id="m-69" dur="8" oct="3" pname="a"/>
                                    <note xml:id="m-73" dur="8" oct="3" pname="b"/>
                                    <note xml:id="m-74" dur="8" oct="4" pname="c"/>
                                    <note xml:id="m-75" dur="8" oct="3" pname="b"/>
                                </beam>
                                <beam xml:id="m-77" staff="1" place="below">
                                    <note xml:id="m-76" dur="8" oct="4" pname="c" staff="1"/>
                                    <note xml:id="m-78" dur="8" oct="4" pname="d" staff="1"/>
                                    <note xml:id="m-79" dur="8" oct="4" pname="e" staff="1"/>
                                    <note xml:id="m-80" dur="8" oct="4" pname="d" staff="1"/>
                                </beam>
                            </layer>
                            <layer xml:id="m-71" n="2">
                                <note xml:id="m-72" dur="1" oct="2" pname="f"/>
                            </layer>
                        </staff>
                        <slur curvedir="mixed" xml:id="m-81" endid="#m-80" layer="1" staff="2" startid="#m-69"/>
                    </measure>
                </section>
            </score>
        </mdiv>
    </body>
</music>
</mei>
```

There is one problem with `@curvedir="mixed"`:  It could also imply this shape:

grafik

Taken from Behind Bars, p. 321. But Gould lists this as discouraged, and technically it's not possible to achieve that shape with a simple Bézier curve anyway. So I guess it would be O.K. to assume that anything with `@curvedir="mixed"` would render like the "simple" s-shape.

There are more complex cases which however would be virtually impossible to render automatically. Slurs are already hard enough without considering those cases.

grafik

### PR DESCRIPTION
**Pull request by DavidBauer1984**
This PR adds support for s-shaped cross staff slurs and closes #1583 .

S-slurs4

The main changes involve support for the initial rendering of the s-shape and tracking signs in order to deal with collisions from two sides. Furthermore s-shaped slurs can request the addition of staff space when collisions happen from both sides.

Another example:
S-slurs3

<details>
<summary> Show MEI </summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/dev/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/dev/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="5.0.0-dev">
 <meiHead>
  <fileDesc>
   <titleStmt>
   </titleStmt>
   <pubStmt>
   </pubStmt>
  </fileDesc>
  <encodingDesc xml:id="encodingdesc-opxjj4">
   <appInfo xml:id="appinfo-5skeys">
    <application xml:id="application-fwbl7x" isodate="2022-02-07T17:32:26" version="3.9.0-dev-6332b7e">
     <name xml:id="name-i64343">Verovio</name>
     <p xml:id="p-odu21t">Transcoded from MusicXML</p>
    </application>
   </appInfo>
  </encodingDesc>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mt9fp2q">
    <score xml:id="s24ma82">
     <scoreDef xml:id="sjlw3o2">
      <staffGrp xml:id="P2" bar.thru="true">
        <label xml:id="ltqjtcr">Piano</label>
        <instrDef xml:id="ikq4w97" midi.channel="1" midi.instrnum="0" midi.volume="78.00%" />
        <staffDef xml:id="s3psp84" n="2" lines="5" ppq="12">
         <clef xml:id="c36o8y9" shape="G" line="2" />
         <keySig xml:id="k49z14r" sig="2f" />
         <meterSig xml:id="mmm9knh" count="4" sym="common" unit="4" />
        </staffDef>
        <staffDef xml:id="s6v5z9o" n="3" lines="5" ppq="12">
         <clef xml:id="cv3h3kz" shape="F" line="4" />
         <keySig xml:id="k1qgehb" sig="2f" />
         <meterSig xml:id="mxazv3p" count="4" sym="common" unit="4" />
        </staffDef>
        <grpSym xml:id="gunom6r" symbol="brace" />
      </staffGrp>
     </scoreDef>
     <section xml:id="slruy1t">
      <pb xml:id="p7m2ec6" />
      <measure xml:id="mld2utm" n="0">
       <mNum xml:id="m6ana2r" />
       <staff xml:id="syyrfcl" n="2">
        <layer xml:id="lfq4tfc" n="1">
         <note xml:id="nybrrhe" dur.ppq="12" dur="4" oct="5" pname="f" stem.dir="down" />
        </layer>
       </staff>
       <staff xml:id="sk4d8l7" n="3">
        <layer xml:id="lnpm7xs" n="5">
         <rest xml:id="rb9qck1" dur.ppq="12" dur="4" />
        </layer>
       </staff>
       <tempo xml:id="tz1im66" place="above" staff="1" tstamp="1.000000" xml:lang="it" midi.bpm="92.000000">
        <rend xml:id="rhokuix" fontweight="bold">Andante</rend>
        <lb xml:id="lxzh4ev" />Con molto espress.<lb xml:id="lpim9c8" />
       </tempo>
       <dynam xml:id="d2wcr7b" place="below" staff="2" tstamp="1.000000" val="49" vgrp="40">p</dynam>
      </measure>
      <measure xml:id="mc3pvmx" n="1">
       <staff xml:id="ssyjpae" n="2">
        <layer xml:id="l91ia7j" n="1">
         <note xml:id="nl8zhe3" dur.ppq="12" dur="4" oct="5" pname="f" stem.dir="down" />
         <note xml:id="ncdycm8" dur.ppq="12" dur="4" oct="5" pname="f" stem.dir="down" />
         <note xml:id="nht8ta0" dots="1" dur.ppq="18" dur="4" oct="6" pname="d" stem.dir="down" />
         <note xml:id="n1usd42" dur="8" oct="6" pname="f" grace="unacc" stem.dir="up" stem.mod="1slash" />
         <note xml:id="ngb4o9q" dur.ppq="6" dur="8" oct="6" pname="e" stem.dir="down" accid.ges="f" />
        </layer>
       </staff>
       <staff xml:id="stvzh8e" n="3">
        <layer xml:id="lel7fuo" n="5">
         <beam xml:id="bx3mjr9">
          <note xml:id="n5q7mei" dur.ppq="6" dur="8" oct="2" pname="f" stem.dir="down" />
          <note xml:id="nelbrux" dur.ppq="6" dur="8" oct="3" pname="f" stem.dir="down" />
          <chord xml:id="cwfs2ac" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="nrvg6s3" oct="3" pname="a" />
           <note xml:id="ncjxq6f" oct="4" pname="c" />
           <note xml:id="n3z2nb5" oct="4" pname="e" accid.ges="f" />
          </chord>
          <note xml:id="nt84i2m" dur.ppq="6" dur="8" oct="3" pname="f" stem.dir="down" />
         </beam>
         <beam xml:id="bcti8di">
          <note xml:id="nimfxo5" dur.ppq="6" dur="8" oct="2" pname="b" stem.dir="down" accid.ges="f" />
          <note xml:id="n9izbg" dur.ppq="6" dur="8" oct="3" pname="f" stem.dir="down" />
          <chord xml:id="c4gfb88" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="nb46i0r" oct="3" pname="b" accid.ges="f" />
           <note xml:id="n3v6iu9" oct="4" pname="d" />
          </chord>
          <note xml:id="n8p6kuq" dur.ppq="6" dur="8" oct="3" pname="f" stem.dir="down" />
         </beam>
        </layer>
       </staff>
       <slur xml:id="slur-1" startid="#n5q7mei" endid="#ngb4o9q" curvedir="mixed" />
      </measure>
      <measure xml:id="md3r1ii" n="2">
       <staff xml:id="s5x6pjj" n="2">
        <layer xml:id="l9yuwep" n="1">
         <beam xml:id="bgczvz4">
          <note xml:id="nirlx0f" dur.ppq="6" dur="8" oct="6" pname="d" stem.dir="down" />
          <note xml:id="n6oz6ep" dur.ppq="6" dur="8" oct="6" pname="c" stem.dir="down" />
          <note xml:id="nu0b41u" dur.ppq="6" dur="8" oct="5" pname="b" stem.dir="down" accid.ges="f" />
          <note xml:id="nnci5z4" dur.ppq="3" dur="16" oct="6" pname="d" stem.dir="down" />
          <note xml:id="ns980nr" dur.ppq="3" dur="16" oct="6" pname="c" stem.dir="down" />
         </beam>
         <note xml:id="ncnojc2" dur.ppq="12" dur="4" oct="5" pname="b" stem.dir="down" accid.ges="f" />
         <rest xml:id="rq55ltv" dur.ppq="6" dur="8" />
         <note xml:id="n4n19qp" dur.ppq="6" dur="8" oct="5" pname="b" stem.dir="down" accid.ges="f" />
        </layer>
       </staff>
       <staff xml:id="srs4i6k" n="3">
        <layer xml:id="l7u3r1i" n="5">
         <beam xml:id="bu4k1kl">
          <note xml:id="n1ya64q" dur.ppq="6" dur="8" oct="2" pname="f" stem.dir="down" accid="s" />
          <note xml:id="n9erbnw" dur.ppq="6" dur="8" oct="3" pname="f" stem.dir="down" accid="s" />
          <chord xml:id="ce57d28" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="n4eq2a3" oct="3" pname="a" />
           <note xml:id="n782o0p" oct="4" pname="c" />
           <note xml:id="nn8gd8i" oct="4" pname="e" accid.ges="f" />
          </chord>
          <note xml:id="n8orphf" dur.ppq="6" dur="8" oct="3" pname="f" stem.dir="down" accid.ges="s" />
         </beam>
         <beam xml:id="b2k6wi">
          <note xml:id="nca8sya" dur.ppq="6" dur="8" oct="2" pname="g" stem.dir="down" />
          <note xml:id="nktab00" dur.ppq="6" dur="8" oct="3" pname="g" stem.dir="down" />
          <chord xml:id="c4b08sr" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="n13hqcx" oct="3" pname="b" accid.ges="f" />
           <note xml:id="nvehpte" oct="4" pname="d" />
          </chord>
          <note xml:id="nrm1xa5" dur.ppq="6" dur="8" oct="3" pname="g" stem.dir="down" />
         </beam>
        </layer>
       </staff>
       <slur xml:id="s1gy797" startid="#nirlx0f" endid="#ns980nr" curvedir="above" />
       <slur xml:id="slur-2" startid="#nirlx0f" endid="#nrm1xa5" curvedir="mixed" />
       <mordent xml:id="mgabgnq" staff="2" startid="#nirlx0f" form="upper" />
       <turn xml:id="tdzxgh6" staff="2" startid="#nu0b41u" form="upper" />
      </measure>
      <measure xml:id="mbtk18z" n="3">
       <staff xml:id="slzh434" n="2">
        <layer xml:id="ltuq1nu" n="1">
         <rest xml:id="re6smu9" dur.ppq="6" dur="8" />
         <beam xml:id="btptg8l">
          <chord xml:id="cqp2e5x" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="n7qwujf" oct="5" pname="g" />
           <note xml:id="no6jxc2" oct="5" pname="b" accid.ges="f" />
          </chord>
          <chord xml:id="c1qa2z4" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="nsbmtoj" oct="5" pname="f" />
           <note xml:id="n7x673" oct="5" pname="a" />
          </chord>
          <chord xml:id="cr6xxl4" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="n7mhwez" oct="5" pname="e" accid.ges="f" />
           <note xml:id="nnz9ll6" oct="5" pname="g" />
          </chord>
         </beam>
         <beam xml:id="b60us1n">
          <chord xml:id="ctntath" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="nj03nu4" oct="5" pname="d" />
           <note xml:id="nm2wzgm" oct="5" pname="f" />
          </chord>
          <chord xml:id="c8au8eu" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="ns1stih" oct="4" pname="f" />
           <note xml:id="na4whja" oct="5" pname="d" />
          </chord>
          <chord xml:id="cdi97y" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="nr4f3qa" oct="4" pname="g" />
           <note xml:id="n24imr0" oct="5" pname="e" accid.ges="f" />
          </chord>
          <chord xml:id="co06yad" dur.ppq="6" dur="8" stem.dir="down">
           <note xml:id="nz2r7s7" oct="4" pname="e" accid.ges="f" />
           <note xml:id="nocnqu7" oct="5" pname="c" />
          </chord>
         </beam>
        </layer>
       </staff>
       <staff xml:id="smdyxts" n="3">
        <layer xml:id="lx8cf5" n="5">
         <chord xml:id="cbt0ndk" dur.ppq="12" dur="4" stem.dir="up">
          <note xml:id="n40sruv" oct="2" pname="e" accid.ges="f" />
          <note xml:id="nm7o6yi" oct="3" pname="e" accid.ges="f" />
         </chord>
         <rest xml:id="rvlkeuk" dur.ppq="12" dur="4" />
         <note xml:id="nmnb74a" dur.ppq="12" dur="4" oct="2" pname="b" stem.dir="up" accid.ges="f" />
         <note xml:id="ngamc6r" dur.ppq="12" dur="4" oct="3" pname="e" stem.dir="down" accid.ges="f" />
        </layer>
       </staff>
       <dynam xml:id="dnqt0fr" place="below" staff="2" tstamp="1.000000" val="96" vgrp="40">f</dynam>
       <slur xml:id="sh2hh2q" startid="#n7qwujf" endid="#n7mhwez" curvedir="above" />
       <slur xml:id="suxlksz" startid="#nj03nu4" endid="#nz2r7s7" curvedir="above" />
       <slur xml:id="slur-3" startid="#cbt0ndk" endid="#co06yad" curvedir="mixed" />
      </measure>
      <sb xml:id="spnkudk" />
      <measure xml:id="mz3gb98" n="4">
       <staff xml:id="sk5zd08" n="2">
        <layer xml:id="lx028vm" n="1">
         <beam xml:id="by3r0zw">
          <chord xml:id="cv1tuo4" dur.ppq="6" dur="8" stem.dir="up">
           <note xml:id="nn2oj6r" oct="4" pname="f" />
           <note xml:id="nc3ghso" oct="5" pname="d" />
          </chord>
          <chord xml:id="craqhwy" dur.ppq="6" dur="8" stem.dir="up">
           <note xml:id="negb57e" oct="4" pname="d" />
           <note xml:id="nhpcvmx" oct="4" pname="b" accid.ges="f" />
          </chord>
          <chord xml:id="cum46zy" dur.ppq="6" dur="8" stem.dir="up">
           <note xml:id="ne20h1p" oct="4" pname="e" accid.ges="f" />
           <note xml:id="nj8pex8" oct="5" pname="c" />
          </chord>
          <chord xml:id="cjxaksi" dur.ppq="6" dur="8" stem.dir="up">
           <note xml:id="n5lanwp" oct="4" pname="c" />
           <note xml:id="n4b8683" oct="4" pname="a" />
          </chord>
         </beam>
         <beam xml:id="bhmxzp6">
          <chord xml:id="cioeki0" dur.ppq="6" dur="8" stem.dir="up">
           <note xml:id="nmc6fbe" oct="4" pname="d" />
           <note xml:id="na12ngr" oct="4" pname="b" accid.ges="f" />
          </chord>
          <chord xml:id="cipufpj" dur.ppq="6" dur="8" stem.dir="up">
           <note xml:id="nvaduar" oct="4" pname="f" />
           <note xml:id="nsoh23t" oct="4" pname="a" />
          </chord>
          <chord xml:id="ccdp7jp" dur.ppq="6" dur="8" stem.dir="up">
           <note xml:id="n4qblfg" oct="4" pname="e" accid.ges="f" />
           <note xml:id="nypyb5v" oct="4" pname="g" />
          </chord>
          <chord xml:id="cwcahm4" dur.ppq="6" dur="8" stem.dir="up">
           <note xml:id="n34lggy" oct="4" pname="d" />
           <note xml:id="nv9w9iy" oct="4" pname="f" />
          </chord>
         </beam>
        </layer>
       </staff>
       <staff xml:id="sd8hy4w" n="3">
        <layer xml:id="lpgo9l7" n="5">
         <note xml:id="ngsbsuo" dur.ppq="12" dur="4" oct="3" pname="f" stem.dir="down" />
         <note xml:id="n1w6xkf" dur.ppq="12" dur="4" oct="2" pname="f" stem.dir="up" />
         <note xml:id="nkht8ni" dur.ppq="12" dur="4" oct="2" pname="b" stem.dir="up" accid.ges="f" />
         <rest xml:id="rro8xxx" dur.ppq="12" dur="4" />
        </layer>
       </staff>
       <slur xml:id="sovccmb" startid="#nn2oj6r" endid="#n1w6xkf" curvedir="mixed" />
       <slur xml:id="s9d5me1" startid="#nkht8ni" endid="#n34lggy" curvedir="mixed" />
      </measure>
      <measure xml:id="m19ebou" n="5">
       <staff xml:id="s64sep5" n="2">
        <layer xml:id="lgmeiz9" n="1">
         <rest xml:id="r64ma3p" dur.ppq="6" dur="8" />
         <beam xml:id="bwqam31">
          <chord xml:id="cybz783" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="am8c47e" artic="stacc" />
           <note xml:id="njlrtne" oct="4" pname="e" accid.ges="f" />
           <note xml:id="ngd1r2d" oct="4" pname="f" />
          </chord>
          <chord xml:id="cbf25od" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="a60ja7g" artic="stacc" />
           <note xml:id="n3jvj91" oct="4" pname="e" accid.ges="f" />
           <note xml:id="nr7th50" oct="4" pname="f" />
          </chord>
          <chord xml:id="c43au6b" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="a9qnc3a" artic="stacc" />
           <note xml:id="na7ncto" oct="4" pname="e" accid.ges="f" />
           <note xml:id="nn7j1bk" oct="4" pname="f" accid="s" />
          </chord>
         </beam>
         <beam xml:id="bdcxti2">
          <chord xml:id="cke7zat" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="as0mc3r" artic="stacc" />
           <note xml:id="naehw4u" oct="4" pname="e" accid.ges="f" />
           <note xml:id="ncbixts" oct="4" pname="g" />
          </chord>
          <chord xml:id="caoonlp" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="apjr6z4" artic="stacc" />
           <note xml:id="nxvz6iy" oct="4" pname="e" accid.ges="f" />
           <note xml:id="nc64eks" oct="4" pname="g" />
          </chord>
          <chord xml:id="ciaypkc" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="amjjohl" artic="stacc" />
           <note xml:id="nsv891l" oct="4" pname="e" accid.ges="f" />
           <note xml:id="n8c4j8n" oct="4" pname="a" />
          </chord>
          <chord xml:id="ct9w3ae" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="aldebjb" artic="stacc" />
           <note xml:id="nxfutz8" oct="4" pname="e" accid.ges="f" />
           <note xml:id="npaboar" oct="4" pname="a" />
          </chord>
         </beam>
        </layer>
       </staff>
       <staff xml:id="snjsega" n="3">
        <layer xml:id="ltv36ck" n="5">
         <beam xml:id="bsjm5bo">
          <note xml:id="nx8w0wd" dur.ppq="6" dur="8" oct="2" pname="f" stem.dir="down">
           <artic xml:id="a2899vw" artic="stacc" />
          </note>
          <chord xml:id="cghmwc7" dur.ppq="6" dur="8" stem.dir="down">
           <artic xml:id="awgdk7k" artic="stacc" />
           <note xml:id="nrkpk7n" oct="3" pname="f" />
           <note xml:id="nc0x34l" oct="3" pname="a" />
          </chord>
          <chord xml:id="ct2w9vg" dur.ppq="6" dur="8" stem.dir="down">
           <artic xml:id="adrsmaq" artic="stacc" />
           <note xml:id="n8i29og" oct="3" pname="f" />
           <note xml:id="nci7ndi" oct="3" pname="a" />
          </chord>
          <chord xml:id="c4r41d2" dur.ppq="6" dur="8" stem.dir="down">
           <artic xml:id="a9l1hz4" artic="stacc" />
           <note xml:id="nw9rv54" oct="3" pname="f" />
           <note xml:id="nkumm8n" oct="3" pname="a" />
          </chord>
         </beam>
         <beam xml:id="b7hlwdj">
          <chord xml:id="cd3b6yc" dur.ppq="6" dur="8" stem.dir="down">
           <artic xml:id="a7ke32u" artic="stacc" />
           <note xml:id="n55toqs" oct="3" pname="f" />
           <note xml:id="ncykbfi" oct="3" pname="b" accid.ges="f" />
          </chord>
          <chord xml:id="cnpyole" dur.ppq="6" dur="8" stem.dir="down">
           <artic xml:id="ad1hmq" artic="stacc" />
           <note xml:id="ny2isfb" oct="3" pname="f" />
           <note xml:id="n9qayn1" oct="3" pname="b" accid.ges="f" />
          </chord>
          <chord xml:id="cvo1vkv" dur.ppq="6" dur="8" stem.dir="down">
           <artic xml:id="ae8on7i" artic="stacc" />
           <note xml:id="n9o54nt" oct="3" pname="f" />
           <note xml:id="noqd0hj" oct="4" pname="c" />
          </chord>
          <chord xml:id="cyvfk42" dur.ppq="6" dur="8" stem.dir="down">
           <artic xml:id="acbiavk" artic="stacc" />
           <note xml:id="nks7v4f" oct="3" pname="f" />
           <note xml:id="nmkc20c" oct="4" pname="c" />
          </chord>
         </beam>
        </layer>
       </staff>
       <slur xml:id="slur-4" startid="#nx8w0wd" endid="#ct9w3ae" curvedir="mixed" />
       <hairpin xml:id="hyhngs0" staff="2" tstamp="1.000000" tstamp2="0m+4.5000" form="cres" place="below" vgrp="75" />
      </measure>
      <measure xml:id="moetc9d" n="6">
       <staff xml:id="scrtfao" n="2">
        <layer xml:id="lut7pr5" n="1">
         <rest xml:id="rwdfypx" dur.ppq="6" dur="8" />
         <beam xml:id="bornmsz">
          <chord xml:id="c2ofh1b" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="aa0vgig" artic="stacc" />
           <note xml:id="nl0xcja" oct="4" pname="d" />
           <note xml:id="nhq6dhs" oct="4" pname="f" />
           <note xml:id="npq0flw" oct="4" pname="b" accid.ges="f" />
          </chord>
          <chord xml:id="cadkf4b" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="akgkyli" artic="stacc" />
           <note xml:id="nkzvic7" oct="4" pname="d" />
           <note xml:id="n9010h8" oct="4" pname="f" />
           <note xml:id="n11hu30" oct="4" pname="b" accid.ges="f" />
          </chord>
          <chord xml:id="coei6ms" dur.ppq="6" dur="8" stem.dir="up">
           <artic xml:id="a7kcio4" artic="stacc" />
           <note xml:id="nphawbt" oct="4" pname="d" />
           <note xml:id="ns8l78u" oct="4" pname="f" />
           <note xml:id="nrejh8k" oct="4" pname="b" accid.ges="f" />
          </chord>
         </beam>
         <chord xml:id="cvfaakb" dur.ppq="12" dur="4" stem.dir="down">
          <note xml:id="nngs3ij" oct="4" pname="b" accid.ges="f" />
          <note xml:id="npzux72" oct="5" pname="d" />
          <note xml:id="nh0tj6c" oct="5" pname="f" />
          <note xml:id="ndulpg7" oct="5" pname="b" accid.ges="f" />
         </chord>
        </layer>
       </staff>
       <staff xml:id="sj227eg" n="3">
        <layer xml:id="lhlr127" n="5">
         <beam xml:id="bz1o4h6">
          <note xml:id="n9d6j78" dur.ppq="6" dur="8" oct="2" pname="b" stem.dir="down" accid.ges="f">
           <artic xml:id="afadic8" artic="stacc" />
          </note>
          <note xml:id="nrvmwim" dur.ppq="6" dur="8" oct="3" pname="b" stem.dir="down" accid.ges="f">
           <artic xml:id="a9qe8ec" artic="stacc" />
          </note>
          <note xml:id="nlyvw03" dur.ppq="6" dur="8" oct="3" pname="b" stem.dir="down" accid.ges="f">
           <artic xml:id="a8zhqji" artic="stacc" />
          </note>
          <note xml:id="nvgfsyi" dur.ppq="6" dur="8" oct="3" pname="b" stem.dir="down" accid.ges="f">
           <artic xml:id="ado42qn" artic="stacc" />
          </note>
         </beam>
         <chord xml:id="cr6oqqw" dur.ppq="12" dur="4" stem.dir="up">
          <note xml:id="nj4rrx8" oct="1" pname="b" accid.ges="f" />
          <note xml:id="ncs3i3l" oct="2" pname="d" />
          <note xml:id="n7s77ma" oct="2" pname="f" />
          <note xml:id="nlxx2s8" oct="2" pname="b" accid.ges="f" />
         </chord>
        </layer>
       </staff>
       <slur xml:id="slur-5" startid="#n9d6j78" endid="#coei6ms" curvedir="mixed" />
       <dynam xml:id="dtpo5z7" place="below" staff="2" tstamp="3.000000" val="96" vgrp="40">f</dynam>
       <arpeg xml:id="aahhbkw" plist="#cvfaakb #cr6oqqw" />
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```
</details>

### CODE REVIEW

File: include/vrv/slur.h [hunk 1]
```diff
@@ -191,13 +253,17 @@ class Slur : public ControlElement,
...
 private:
     /**
-     * The drawing curve direction.
-     * This is calculated only when start - end points are on the same system. Otherwise
-     * it is left unset. This also means that it is reset only in ResetDrawing and not when
-     * the alignment is reset. The reason is because we want to preserve the value when the
-     * document is cast-off.
+     * The drawing curve direction
+     * This is calculated in the PrepareSlurs functor and contains an additional distinction
+     * for s-shaped slurs / mixed direction: whether the slur goes upwards or downwards
```
Comment (lpugin) [inline include/vrv/slur.h:258]: * Why removing the entire previous code comment?
* Maybe we can drop the `SlurCurveDirection` enum if [this](https://github.com/music-encoding/music-encoding/issues/912) is agreed?

**Review by lpugin — COMMENTED** [review]
Maybe a code simplification (for later?) would be to store have a  `FloatingCurvePositioner`?

File: include/vrv/slur.h (hunk 1 above)
Comment (DavidBauer1984) [reply include/vrv/slur.h]: - The previous code comment was not accurate anymore: the slur (even if it is broken) always relies on `m_drawingCurveDir` once `PrepareSlurs` was evaluated. This stores the drawing direction of the whole slur. The direction of the broken pieces is calculated via `Slur::CalcDrawingCurveDir`.
- Exactly, `SlurCurveDirection` should be temporary in the code as long as we cannot distinguish the S-slur direction via `curvature_CURVEDIR`. Once we have the extension on LibMEI, we can kick it out again.

**Comment by lpugin** [comment]
Could you double check test slur-019.mei

Before
image
After
image

**Comment by DavidBauer1984** [comment]
My bad. I moved `PrepareSlurs` into vertical layout (seemed to fit better), but it has to run before systems are cast off.

**Comment by DavidBauer1984** [comment]
My bad. I moved `PrepareSlurs` into vertical layout (seemed to fit better), but it has to run before systems are cast off.

**Comment by craigsapp** [comment]
Screen Shot 2022-03-11 at 7 37 02 AM

**Pull request merged**